### ISSUE TITLE
React UNSAFE_ methods not passing

### ISSUE THREAD

**Issue by JohnPhamous** [microsoft/tslint-microsoft-contrib#887]
### Bug Report

-   **`tslint-microsoft-contrib` version**: 6.2.0
-   **TSLint version**: 5.10.0
-   **TypeScript version**: 3.5.3
-   **Running TSLint via**: CLI webpack prod build, VS Code

#### TypeScript code being linted

```tsx
public UNSAFE_componentWillReceiveProps(nextProps: IOnboardReleaseFormProps): void {
   ...
  }
```

#### Actual behavior

Throws this error: `Method name does not match /^[a-z][\w\d]+$/: UNSAFE_componentWillReceiveProps (function-name)`

#### Expected behavior

[React 16.9.0](https://reactjs.org/blog/2019/08/08/react-v16.9.0.html) added `UNSAFE_*` methods. The regex currently being used doesn't allow for the `UNSAFE_` prefix.

#### Proposed Changes

Not sure how yall want to handle this since it's a library dependent change. What do yall think of changing the regex to `([a-z]|UNSAFE)[\w\d]+`?

**Comment by JoshuaKGoldberg** [comment]
👋 @JohnPhamous good idea! Fortunately, the `function-name` rule supports overriding the regular expressions used already:

https://github.com/microsoft/tslint-microsoft-contrib/blob/bfc28e95dde1fab92aa457e79d338fb03b81d7cc/src/functionNameRule.ts#L73
